sdk/src/Core/Amazon.Util/AWSSDKUtils.cs

```diff
@@ -1056,12 +1056,31 @@ public static string UrlEncode(int rfcNumber, string data, bool path)
                 }
                 else
                 {
-                    encoded.Append("%").Append(string.Format(CultureInfo.InvariantCulture, "{0:X2}", (int)symbol));
+                    encoded.Append('%');
+
+                    // Break apart the byte into two four-bit components and
+                    // then convert each into their hexadecimal equivalent.
+                    byte b = (byte)symbol;
+                    int hinibble = (b >> 4) & 0xF;
```

Comment:
Wouldn't shifting 4 bits to the right make the bitwise AND operator unnecessary for the hinibble since we're isolating those bits anyways? I think you can remove the bitwise AND here.